### PR TITLE
AzureSentinel update incidents

### PR DESCRIPTION
**Pull request by dorschw**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35217

## Description
Adding `Labels` and `Assignee Email` to get_update_incident_request_data

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 1]
```diff
@@ -196,6 +197,8 @@ Updates a single incident in Azure Sentinel.
 | status | The incident status. | Optional | 
 | classification | The reason the incident was closed. Required when updating the status to Closed. Possible values are:  BenignPositive, FalsePositive, TruePositive, Undetermined | Optional | 
 | classification_reason | The classification reason the incident was closed with. Required when updating the status to Closed and the classification is determined. Possible values are:  InaccurateData, IncorrectAlertLogic, SuspiciousActivity, SuspiciousButExpected | Optional | 
+| assignee_email | Email of the incident assignee. Note that the API field updated is `owner.email` | Optional |
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:200]: Make sure this matches the description in the yaml.
```suggestion
| assignee_email | Email of the incident assignee. Note that the updated API field is `owner.email` | Optional | 
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 2]
```diff
@@ -35,14 +35,15 @@
 
 
 class Client:
-    def __init__(self, self_deployed, refresh_token, auth_and_token_url, enc_key, redirect_uri, auth_code,
-                 subscription_id, resource_group_name, workspace_name, verify, proxy):
+    def __init__(self, self_deployed: bool, refresh_token: str, auth_and_token_url: str, enc_key: str,
+                 redirect_uri: str, auth_code: str, subscription_id: str, resource_group_name: str, workspace_name: str,
+                 verify: bool, proxy: bool):
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:40]: nice

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 3]
```diff
@@ -113,9 +114,9 @@ def incident_data_to_demisto_format(inc_data):
         'Status': properties.get('status'),
         'AssigneeName': properties.get('owner', {}).get('assignedTo'),
         'AssigneeEmail': properties.get('owner', {}).get('email'),
-        'Label': [{
-            'Name': label.get('name'),
-            'Type': label.get('type')
+        'Labels': [{
+            'Name': label.get('labelName'),
+            'Type': label.get('labelType')
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:119]: What's the expected output in the yaml?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 4]
```diff
@@ -134,40 +135,53 @@ def incident_data_to_demisto_format(inc_data):
...
 
-    inc_data = {
-        'etag': result.get('etag'),
+    existing_labels = demisto.get(fetched_incident_data, 'properties.labels')
+    if not labels:  # not provided as arg
+        labels_formatted = existing_labels
+
+    else:
+        labels_formatted = [{"labelName": label, "labelType": "User"}
+                            for label in argToList(labels) if label]  # labels can not be blank
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:168]: what about the existing labels? does this override them?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 5]
```diff
@@ -134,40 +135,53 @@ def incident_data_to_demisto_format(inc_data):
...
+    incident_data = {
+        'etag': fetched_incident_data.get('etag'),
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:170]: new field in the API?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 6]
```diff
@@ -134,40 +135,53 @@ def incident_data_to_demisto_format(inc_data):
...
         'properties': {
             'title': title,
             'description': description,
             'severity': severity,
             'status': status,
             'classification': classification,
-            'classificationReason': classification_reason
+            'classificationReason': classification_reason,
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:177]: new field in the API?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 7]
```diff
@@ -74,6 +74,12 @@ configuration:
   name: proxy
   required: false
   type: 8
+- display: Timeout
+  name: timeout
+  additionalinfo: in seconds
+  hidden: false
+  required: false
+  type: 0
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:82]: You didn't implement this parameter. Should it be removed?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 8]
```diff
@@ -289,7 +295,7 @@ script:
       - High
       - Medium
       - Low
-      - Informal
+      - Informational
```
Comment (DeanArbel) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:298]: Add this to the RN

**Review by DeanArbel — CHANGES_REQUESTED** [review]
Nice, see my comments.

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 3 above)
Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:119]: The yaml states a `Label` output, I'll change it. 

In `INCIDENT_HEADERS` (used for human-readable output), the header is  `Labels` though. Should they match or is it a BW compatibility issue if we change it now?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 4 above)
Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:168]: Yes, as that's what happens on a regular API call.

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml (hunk 7 above)
Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:82]: yup

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 5 above)
Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:170]: just a variable name change

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 3 above)
Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py]: `Label` it is

**Review by DeanArbel — APPROVED** [review]
@amshamah419 Can you please give this PR a look?

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 9]
```diff
@@ -124,6 +124,7 @@ Gets a list of incidents from Azure Sentinel.
 ##### Base Command
 
 `azure-sentinel-list-incidents`
+`azure-sentinel-update-incident incident_id=1234567 severity=Informational assignee_email=example@example.com`
```
Comment (roysagi) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:127]: this should not appear here - this only the base command.
you might want to generate a new command example and put its inputs below.

Comment (dorschw) [reply Packs/AzureSentinel/Integrations/AzureSentinel/README.md:127]: thanks

**Pull request merged**